x-pack/plugins/ml/public/url_generator.ts

```diff
@@ -64,41 +48,280 @@ export class MlUrlGenerator implements UrlGeneratorsDefinition<typeof ML_APP_URL
 
   public readonly id = ML_APP_URL_GENERATOR;
 
-  public readonly createUrl = async ({ page, ...params }: MlUrlGeneratorState): Promise<string> => {
-    if (page === 'explorer') {
-      return this.createExplorerUrl(params);
+  public readonly createUrl = async (mlUrlGeneratorState: MlUrlGeneratorState): Promise<string> => {
+    switch (mlUrlGeneratorState.page) {
+      case ML_PAGES.ANOMALY_DETECTION_JOBS_MANAGE:
+        return this.createAnomalyDetectionJobManagementUrl(
+          mlUrlGeneratorState as AnomalyDetectionUrlState
+        );
+      case ML_PAGES.ANOMALY_EXPLORER:
+        return this.createExplorerUrl(mlUrlGeneratorState as ExplorerUrlState);
+      case ML_PAGES.ANOMALY_DETECTION_CREATE_JOB_SELECT_TYPE:
+        return this.createAnomalyDetectionCreatJobSelectType(
+          mlUrlGeneratorState as MlGenericUrlState
+        );
+      case ML_PAGES.SINGLE_METRIC_VIEWER:
+        return this.createSingleMetricViewerUrl(mlUrlGeneratorState as TimeSeriesExplorerUrlState);
+      case ML_PAGES.DATA_FRAME_ANALYTICS_JOBS_MANAGE:
+        return this.createDataframeAnalyticsUrl(mlUrlGeneratorState as DataFrameAnalyticsUrlState);
+      case ML_PAGES.DATA_FRAME_ANALYTICS_EXPLORATION:
+        return this.createDataframeAnalyticsExplorationUrl(
+          mlUrlGeneratorState as DataFrameAnalyticsExplorationUrlState
+        );
+      case ML_PAGES.DATA_VISUALIZER:
+        return this.createDataVisualizerUrl(mlUrlGeneratorState as DataVisualizerUrlState);
```

Comment:
these 3 are the same and so cold fall through to one case

Reply:
Updated here https://github.com/elastic/kibana/pull/75696/commits/4d6a5a0e30e0046121d1febced6b0399da72b25a